jmxtrans-output/jmxtrans-output-kafka/src/main/java/com/googlecode/jmxtrans/model/output/kafka/KafkaWriterFactory.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * The MIT License
+ * Copyright © 2010 JmxTrans team
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans.model.output.kafka;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.googlecode.jmxtrans.model.OutputWriterFactory;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@EqualsAndHashCode(exclude = {"objectMapper"})
+public class KafkaWriterFactory implements OutputWriterFactory<KafkaWriter2> {
+    @Nonnull
+    @Getter @JsonIgnore
+    private final ObjectMapper objectMapper;
+    @Nonnull
+    @Getter
+    private final Map<String, Object> producerConfig;
+    @Nonnull
+    @Getter
+    private final String topic;
+    @Nonnull
+    @Getter
+    private final ResultSerializer resultSerializer;
+
+    private static final String STRING_SERIALIZER_CLASS = StringSerializer.class.getName();
+
+    @JsonCreator
+    public KafkaWriterFactory(
+            @JsonProperty("producerConfig") @Nonnull Map<String, Object> producerConfig,
+            @JsonProperty("topic") @Nonnull String topic,
+            @JsonProperty("resultSerializer") ResultSerializer resultSerializer) {
```

Comment:
Exposing the resultserializer allows to easily change the serialization strategy (at the moment Default/Detail, but could also be custom one).

Yet I agree I can make it bit more backward compatible.